### PR TITLE
perlPackages.DBDSQLite: 1.66 -> 1.70

### PR DESCRIPTION
**Pull request by trofi**
Among other things fixes test failures against sqlite-3.37.

### CODE REVIEW

**Comment by trofi** [comment]
Converting to draft. Looks like `1.70` is not enough. Tests don't fail deterministically.

**Comment by trofi** [comment]
Pulled upstream testsuite tweak. While at it rebased against `staging-next`.

File: pkgs/top-level/perl-packages.nix [hunk 1]
```diff
@@ -6091,6 +6091,13 @@ let
     patches = [
       # Support building against our own sqlite.
       ../development/perl-modules/DBD-SQLite/external-sqlite.patch
+
+      # Pull upstream fix for test failures against sqlite-3.37.
+      (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/top-level/perl-packages.nix:6096]: Can we update to https://github.com/DBD-SQLite/DBD-SQLite/releases/tag/1.71_01 instead?

Comment (jonringer) [reply pkgs/top-level/perl-packages.nix:6096]: The perl updates pull from cpan, not github

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 148574 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary>  <ul>    <li>perlPackages.DBDSQLite</li>  </ul></details>

**Pull request merged**